### PR TITLE
[SPACES] Spaces available only for fresh accounts

### PR DESCRIPTION
**Pull request by abelgardep**
## Related Issues
Implements https://github.com/owncloud/android/issues/3906

Library PR (if needed):

- [ ] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt [hunk 1]
```diff
@@ -237,7 +238,7 @@ abstract class DrawerActivity : ToolbarActivity() {
     private fun setSpacesVisibilityBottomBar(uiResult: UIResult<OCCapability>) {
         if (uiResult is UIResult.Success) {
             val capabilities = uiResult.data
-            if (capabilities?.isSpacesAllowed() == true) {
+            if (AccountUtils.isFeatureSpacesAllowedForAccount(baseContext, account) && capabilities?.isSpacesAllowed() == true) {
```
Comment (JuancaG05) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt:241]: I see this condition is used several times along the code... Could we move it to a method in a common place?

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Some comments here 👍  Good job!

**Review by JuancaG05 — APPROVED** [review]
Ready to QA @jesmrec 🚀

**Comment by jesmrec** [comment]
AFAICS, nothing to test here. Just a new flag and logic to manage it. It will be tested together with the UI notifications for users to re-attach legacy accounts. CI is green, OK from my side unless you point me to test something else here.

**Comment by jesmrec** [comment]
## QA

- [X] Add oCIS and oC10 accounts after fresh install -> oCIS accounts shows spaces, oC10 accounts do not show spaces

With oCIS and oC10 account in the device in `master`:

- [X] Update from `master` -> oCIS accounts do not show spaces, oC10 accounts do not show spaces
- [X] Add account after updating ->  oCIS accounts show spaces, oC10 accounts do not show spaces

Approved on my side

**Pull request merged**